include/core/Partition_One.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef PARTITION_ONE_H
+#define PARTITION_ONE_H
+
+#ifdef NGEN_MPI_ACTIVE
+
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+#include <unordered_set>
+#include <FeatureBuilder.hpp>
+#include "features/Features.hpp"
+#include <FeatureCollection.hpp>
+#include "Partition_Data.hpp"
+
+class Partition_One {
+
+    public:
+        Partition_One() {};
+
+        /**
+         * The function that parses geojson::GeoJSON data and build unordered sets of catchment_ids and nexus_ids
+
+         * @param catchment_collection the geojson::GeoJSON data containing all the necessary hydrofabric info
+         */
+        void generate_partition(geojson::GeoJSON& catchment_collection)
+        {
+            for(auto& feature: *catchment_collection)
+            {
+                std::string cat_id = feature->get_id();
+                catchment_ids.emplace(cat_id);
+                std::string nex_id = feature->get_property("toid").as_string();
+                nexus_ids.emplace(nex_id);
+            }
+            partition_data.mpi_world_rank = 0;
+            partition_data.catchment_ids = catchment_ids;
+            partition_data.nexus_ids = nexus_ids;
+        }
+
+    	virtual ~Partition_One(){};
+
+        //PartitionData is a struct
+        PartitionData partition_data;       
+
+    private:
+        int mpi_world_rank;
+        std::unordered_set<std::string> catchment_ids;
+        std::unordered_set<std::string> nexus_ids;
+        std::vector<Tuple> remote_connections;
```

Comment:
These members seem to duplicate the members of `PartitionData`. Why not just use them?

Reply:
These variables are used this way following the example of ```Partition_Parser.hpp```. In this form, it seems the logic in the loop above is simpler. We could use the struct members, but we would seem to use longer variable names with somewhat overuse of the "." operator.

Reply:
Syntactic simplicity isn't a good reason to duplicate data in memory and add extra assignments, as well as just adding extra declarations that anyone reading the code later has to make sense of.

Reply:
Revised as requested.